src/mozilla_taskgraph/transforms/scriptworker/signing.py

```diff
@@ -0,0 +1,158 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import (
+    Schema,
+    optionally_keyed_by,
+    resolve_keyed_by,
+)
+from voluptuous import ALLOW_EXTRA, Any, Optional, Required
+
+SIGNING_FORMATS = ["autograph_gpg"]
+SIGNING_TYPES = ["dep", "release"]
+DETACHED_SIGNATURE_EXTENSION = ".asc"
+
+signing_schema = Schema(
+    {
+        Required("attributes"): {
+            Optional("artifacts"): dict,
+            Required("build-type"): str,
+        },
+        Required("signing"): optionally_keyed_by(
+            "build-type",
+            "level",
+            {
+                Required("format"): optionally_keyed_by(
```

Comment:
It looks like we [sometimes sign with multiple formats at once](https://searchfox.org/mozilla-central/rev/31f5847a4494b3646edabbdd7ea39cb88509afe2/taskcluster/gecko_taskgraph/util/signed_artifacts.py#89). Do you want to handle this now, or is that going to come later?

Reply:
I think that will come later. For now it's easier for me if I just focus on a single project at a time (`firefox-android`). But thanks for the heads up that this will be needed at some point.